src/sortStudents.ts

```diff
@@ -1,16 +1,66 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverage(grades: number[]): number {
+  const gradesSum = grades.reduce((a, b) => a + b, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return gradesSum / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return studentsCopy.sort((studentA, studentB) => (
+        order === 'asc'
+          ? studentA[sortBy].localeCompare(studentB[sortBy])
+          : studentB[sortBy].localeCompare(studentA[sortBy])
+      ));
+
+    case SortType.Age:
+      return studentsCopy.sort((studentA, studentB) => (
+        order === 'asc'
+          ? studentA[sortBy] - studentB[sortBy]
+          : studentB[sortBy] - studentA[sortBy]
+      ));
+
+    case SortType.Married:
+      return studentsCopy.sort((studentA, studentB) => (
+        order === 'asc'
+          ? Number(studentA[sortBy]) - Number(studentB[sortBy])
+          : Number(studentB[sortBy]) - Number(studentA[sortBy])
+      ));
```

Comment:
```suggestion
    case SortType.Age:
    case SortType.Married:
      return studentsCopy.sort((studentA, studentB) => (
        order === 'asc'
          ? Number(studentA[sortBy]) - Number(studentB[sortBy])
          : Number(studentB[sortBy]) - Number(studentA[sortBy])
      ));
```
Good job, only one moment - we can combine this two cases in one

Reply:
Agree.